Purchases/SubscriberAttributes/RCPurchases+SubscriberAttributes.m

```diff
@@ -50,8 +51,16 @@ - (RCSubscriberAttributeDict)unsyncedAttributesByKey {
     return [self.subscriberAttributesManager unsyncedAttributesByKeyForAppUserID:self.appUserID];
 }
 
-- (void)markAttributesAsSynced:(RCSubscriberAttributeDict)syncedAttributes
-                     appUserID:(NSString *)appUserID {
+- (void)markAttributesAsSyncedIfNeeded:(RCSubscriberAttributeDict)syncedAttributes
+                             appUserID:(NSString *)appUserID
+                                 error:(nullable NSError *)error {
+    if (error && !error.successfullySynced) {
```

Comment:
`error.successfullySynced` will always be NO for the postReceipt cases since the `RCSuccessfullySyncedKey` will always be missing from the user info dictionary right?

Reply:
https://github.com/RevenueCat/purchases-ios/pull/194/files#diff-d337c62bf07cca18d1d0e6dce1177732R539 adds it, called from https://github.com/RevenueCat/purchases-ios/pull/194/files#diff-d337c62bf07cca18d1d0e6dce1177732R99

Reply:
ohh, just realized that there's a bug here - if the receipt is posted successfully, the code = 200 and it never goes into the case that adds the subscriber attributes keys. fixing

Reply:
fixed